querybook/server/lib/result_store/stores/file_store.py

```diff
@@ -28,10 +28,11 @@ def __init__(self, uri: str):
 
     def start(self):
         self._chunks_length = 0
+        self._chunks = []
         os.makedirs(self.uri_dir_path, exist_ok=True)
 
     def write(self, data: str):
-        # write each line into csv
+        # Append each line to a list and then dump at once
```

Comment:
dumping at once may cause issue when the result set size to be uploaded is too big. 
A better way could be dumping by chunks, collecting X rows and then dump it.

Reply:
About my PR:

Previously, the write method in FileUploader class was opening the file in append mode and writing data into it for each record of the result set of the query. Which perhaps was talking a lot of time if the result set is huge and is also not a good practice to increase the file operation. 

So to increase the performance, we did the below:
-        Initialised self._chunks as an empty list in start method of FileUploader class.
-        Kept appending each record of the query result to the above list in write method of FileUploader class. 
-        At the end of the process, Converted the list elements to string using join in end method of FileUploader class.
-        Opened the file in write mode and wrote the above string value to file in end method of FileUploader class.

This was tested and we observed that the time taken to write the data to file was in a few seconds.

I appreciate you in suggesting an alternative approach for dumping the data. 

Based on the approach as explained above is there still a need of dumping by chunks as we are using join iterator to convert list to string and then write that to file. 

Please feel free to correct me if I am wrong.

Thanks for your time.

Reply:
thanks for the for detailed explanation.
yes, there is no doubt here that the current way of opening the file and append it for each record doesn't have good performance. 
What I meant was for the extreme case like:
when the file to be uploaded is too big to be fit into the container's allocated memory, you cant just keep appending each row to `self._chunks` and keep it in the memory, but have to write it the file in batches.